lib/credit_card_sanitizer.rb

```diff
@@ -95,15 +96,15 @@ def sanitize!(text, options = {})
 
     text.force_encoding(Encoding::UTF_8)
     text.scrub!('�')
-    regex = options[:exclude_html_tags] ? SKIP_HTML_TAGS : NUMBERS_WITH_LINE_NOISE
     redacted = nil
+
     without_expiration(text) do
-      text.gsub!(regex) do |match|
-        next match if $1 || $2
+      text.gsub!(NUMBERS_WITH_LINE_NOISE) do |match|
+        next match if $1
 
-        candidate = Candidate.new(match, match.tr('^0-9', ''))
+        candidate = Candidate.new(match, match.tr('^0-9', ''), $`, $')
 
-        if valid_numbers?(candidate, options)
+        if valid_numbers?(candidate, options) && valid_context?(candidate, options)
```

Comment:
`valid_context?` will be a faster check than `valid_numbers?`, so I would swap the `&&` order here to do it first ... you can short-circuit out of the Luhn check and credit card company prefix checks, by doing the context check first